conda_smithy/schema.py

```diff
@@ -0,0 +1,1165 @@
+    cases where you need to map enum values to corresponding data structures.
+
+    Args:
+        enum_class (Union[Type[Enum], Type[EnumMeta]]): A single Enum class or a Union of Enum classes representing the keys for the generated model.
+        value_model (BaseModel): A Pydantic model representing the structure of the values.
+        model_name (str): The name of the generated Pydantic model.
+
+    Example usage:
+
+    ```python
+    class MyEnum1(str, Enum):
+        key1 = "key1"
+        key2 = "key2"
+
+    class MyEnum2(str, Enum):
+        key3 = "key3"
+        key4 = "key4"
+
+    class ValueModel(BaseModel):
+        value_field: int
+
+    # Create an instance of PydanticModelGenerator with a single Enum
+    model_generator1 = PydanticModelGenerator(MyEnum1, ValueModel, "MyGeneratedModel1")
+
+    # Create an instance of PydanticModelGenerator with a Union of Enums
+    model_generator2 = PydanticModelGenerator(Union[MyEnum1, MyEnum2], ValueModel, "MyGeneratedModel2")
+
+    # Retrieve the generated models using the __call__ method
+    GeneratedModel1 = model_generator1()
+    GeneratedModel2 = model_generator2()
+    ```
+    """
+
+    def __init__(
+        self,
+        enum_class: Union[Type[Enum], Type[EnumMeta]],
+        value_model: BaseModel,
+        model_name: str,
+    ):
+        self.enum_class = enum_class
+        self.value_model = value_model
+        self.model_name = model_name
+
+    def __call__(self) -> BaseModel:
+        field_definitions = {}
+
+        if hasattr(self.enum_class, "__args__"):
+            # Handle Union types
+            for enum_type in self.enum_class.__args__:
+                if isinstance(enum_type, type) and issubclass(enum_type, Enum):
+                    for enum_item in enum_type:
+                        field_definitions[enum_item.value] = (
+                            Optional[self.value_model],
+                            {"description": f"The {enum_item.value} value"},
+                        )
+        else:
+            # Handle single Enum
+            for enum_item in self.enum_class:
+                field_definitions[enum_item.value] = (
+                    Optional[self.value_model],
+                    {"description": f"The {enum_item.value} value"},
+                )
+
+        return create_model(self.model_name, **field_definitions)
+
+
+#############################################
+######## Choices (Enum/Literals) definitions #########
+#############################################
+
+conda_build_tools = Literal[
+    # will run vanilla conda-build, with system configured / default solver
+    "conda-build",
+    # will run vanilla conda-build, with the classic solver
+    "conda-build+classic",
+    # will run vanilla conda-build, with libmamba solver
+    "conda-build+conda-libmamba-solver",
+    # will run 'conda mambabuild', as provided by boa
+    "mambabuild",
+]
+
+
+class CIservices(str, Enum):
```

Comment:
We are importing `StrEnum` above, but then using `str, Enum` here? I guess we should use here as well (and in the other occurrences).

Reply:
Done in [2cf8f73](https://github.com/conda-forge/conda-smithy/pull/1756/commits/2cf8f7378b25ec6da3f6a03964bbcf1551932b38)